cdap-ui/app/cdap/components/DataPrep/store/DataPrepActionCreator.js

```diff
@@ -120,3 +123,52 @@ export function setWorkspace(workspaceId) {
       });
   });
 }
+
+export function getWorkspaceList(workspaceId) {
+  let namespace = NamespaceStore.getState().selectedNamespace;
+
+  return Rx.Observable.create((observer) => {
+    MyDataPrepApi.getWorkspaceList({ namespace })
+      .subscribe((res) => {
+        if (res.values.length === 0) {
+          observer.onNext([]);
+          observer.onCompleted();
+          return;
+        }
+
+        let workspaceList = sortBy(res.values, ['name']);
+
+        DataPrepStore.dispatch({
+          type: DataPrepActions.setWorkspaceList,
+          payload: {
+            list: workspaceList
+          }
+        });
+
+        if (workspaceId) {
+          // Set active workspace
+          // Check for existance of the workspaceId
+
+          let match = workspaceList.filter((workspace) => workspace.id === workspaceId);
+          let workspaceStream;
+          if (match.length > 0) {
```

Comment:
Can there be duplicate workspaceIDs? Also `setWorkspace` accepts only a workspace Id right? Can we pass array that gets `toString()`ed in the URL to setWorkspace?